hepdata/modules/records/utils/submission.py

```diff
@@ -570,6 +573,31 @@ def process_submission_directory(basepath, submission_file_path, recid,
 
         if len(errors) is 0:
             errors = package_submission(basepath, recid, hepsubmission)
+
+            # Check the size of the upload to ensure it can be converted
+            data_filepath = find_submission_data_file_path(hepsubmission)
+            with prepare_data_folder(data_filepath, 'yaml') as filepaths:
+                input_directory, input_file = filepaths
+                # Create options that look like a worst-case (biggest)
+                # conversions (using yoda-like options as they include rivet
+                # analysis
+                dummy_inspire_id = hepsubmission.inspire_id or '0000000'
+                options = {
+                    'input_format': 'yaml',
+                    'output_format': 'yoda',
+                    'filename': f'HEPData-ins{dummy_inspire_id}-v{hepsubmission.version}-yoda',
+                    'validator_schema_version': '0.1.0',
+                    'hepdata_doi': f'10.17182/hepdata.{recid}.v{hepsubmission.version}',
+                    'rivet_analysis_name': f'ATLAS_2020_I{dummy_inspire_id}'
+                }
+                data_size = get_data_size(data_filepath, options)
+                if data_size > current_app.config['UPLOAD_MAX_SIZE']:
+                    errors["Archive"] = [{
+                        "level": "error",
+                        "message": "Archive is too big for conversion to other formats. (%s bytes would be sent to converter.)"
+                                   % data_size
+                    }]
```

Comment:
It might be useful also to write `UPLOAD_MAX_SIZE` in the error message so that the Uploader can see what to aim for.